### PR TITLE
Fix exit code for command line errors

### LINKED ISSUE #3913: Pytest exit codes for command line errors look incorrect
According to [the documentation](https://docs.pytest.org/en/latest/usage.html), the CLI return value for a bug on the command line should be 4. From what I can tell, all bugs return 2.

```bash
$ pytest -invalid-
pytest: error: unrecognized arguments: -invalid-
  inifile: None
  rootdir: /tmp
$ echo $?
2
$ pytest -v-v
usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: argument -v/--verbose: ignored explicit argument '-v'
$ echo $?
2
```

It looks like interrupting PyTest does return 2, however:
```
$ pytest xgt_test.py 
================================ test session starts ================================
platform linux2 -- Python 2.7.15rc1, pytest-3.7.2, py-1.5.4, pluggy-0.7.1
rootdir: /gems/build/apps/mygems, inifile:
collected 26 items                                                                  

xgt_test.py EEEEEEEEEEE^C
<snip>
$ echo $?
2
```

System configuration:
```
$ pytest --version
This is pytest version 3.7.2, imported from /usr/local/lib/python2.7/dist-packages/pytest.pyc
$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04 LTS"
```

Happy to answer questions, and apologies if I've misunderstood the intended behavior.

### PR DESCRIPTION
**Pull request by crazymerlyn**
Fixes #3913

### CODE REVIEW

File: changelog/3913.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Pytest now returns with correct exit code when called with unknown arguments.
```
Comment (nicoddemus) [inline changelog/3913.bugfix.rst]: Please change to:

> Pytest now returns with correct exit code (EXIT_USAGEERROR, 4) when called with unknown arguments.

**Comment by nicoddemus** [comment]
Oh and I forgot: please target the `features` branch. Although this is a bug fix, it is still changing the outcome of the pytest command, so we better be safe and release this in `3.8`. 👍

**Comment by crazymerlyn** [comment]
Changelog entry fixed, test added and the pull request now targets the features branch.

File: src/_pytest/config/argparsing.py [hunk 2]
```diff
@@ -2,8 +2,13 @@
 import warnings
 import argparse
 
+from gettext import gettext as _
```
[the remark this answers is not in the document]
Comment (nicoddemus) [reply src/_pytest/config/argparsing.py]: Oh right, totally missed that. Weird though, shouldn't flake8 catch that?

[the remark this answers is not in the document]
Comment (crazymerlyn) [reply src/_pytest/config/argparsing.py]: @RonnyPfannschmidt  The default implementation of error function used it and I didn't want to change anything other than the error code.

[the remark this answers is not in the document]
Comment (nicoddemus) [reply src/_pytest/config/argparsing.py]: Oh right, it is used:

```python
self.exit(EXIT_USAGEERROR, _("%(prog)s: error: %(message)s\n") % args)
```

Even though we don't translate pytest messages, I don't mind leaving it in as `gettext` is already being imported by `argparser` anyway. But want me to remove it @RonnyPfannschmidt?

[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply src/_pytest/config/argparsing.py]: @crazymerlyn thanks for the additional context
@nicoddemus thanks for the investigation

im a bit thorn between keeping similarity to imported code and limiting code scope, lets keep it as is for now

**Pull request merged**